src/txn/txn.c

```diff
@@ -381,7 +396,8 @@ __wt_txn_release(WT_SESSION_IMPL *session)
 	 */
 	__wt_txn_release_snapshot(session);
 	txn->isolation = session->isolation;
-	F_CLR(txn, WT_TXN_ERROR | WT_TXN_HAS_ID | WT_TXN_RUNNING);
+	F_CLR(txn, WT_TXN_ERROR | WT_TXN_HAS_ID |
+	    WT_TXN_NAMED_SNAPSHOT | WT_TXN_READONLY | WT_TXN_RUNNING);
```

Comment:
@michaelcahill, the flag not being cleared here is `WT_TXN_AUTOCOMMIT`, and I don't see why (I don't think there's a bug, it's just odd).

`WT_TXN_HAS_SNAPSHOT` was cleared two lines before, by `__wt_txn_release_snapshot`.

Anyway, I'd switch to `txn->flags = 0;` for clarity.